src/legal-officer/transaction-protection/LocContext.tsx

```diff
@@ -0,0 +1,185 @@
+        case "SET_LOC_REQUEST":
+            return { ...state, locRequest: action.locRequest! }
+        case "SET_LOC":
+            return { ...state, loc: action.loc! }
+        case "SET_FUNCTIONS":
+            return {
+                ...state,
+                addMetadata: action.addMetadata!,
+                publishMetadata: action.publishMetadata!,
+                removeMetadata: action.removeMetadata!,
+                changeItemStatus: action.changeItemStatus!
+            }
+        case "ADD_ITEM":
+            return { ...state, locItems: state.locItems.concat(action.locItem!) }
+        case "UPDATE_ITEM":
+            items[itemIndex] = { ...action.locItem!, status: action.status! }
+            return { ...state, locItems: items }
+        case "DELETE_ITEM":
+            items.splice(itemIndex, 1)
+            return { ...state, locItems: items }
+        default:
+            throw new Error(`Unknown type: ${ action.type }`);
+    }
+}
+
+export interface Props {
+    locId: UUID
+    children: JSX.Element | JSX.Element[] | null
+}
+
+export function LocContextProvider(props: Props) {
+    const { api } = useLogionChain();
+    const { openedLocRequests } = useCommonContext();
+    const [ contextValue, dispatch ] = useReducer(reducer, initialContextValue(props.locId));
+
+    useEffect(() => {
+        if (contextValue.locRequest === null && openedLocRequests !== null) {
+            const locRequest = openedLocRequests.find(locRequest => locRequest.id === contextValue.locId.toString());
+            if (locRequest !== undefined) {
+                dispatch({ type: 'SET_LOC_REQUEST', locRequest })
+            }
+        }
+    }, [ contextValue, openedLocRequests ])
+
+    const addMetadataFunction = useCallback((name: string, value: string) => {
+            const locItem: LocItem = {
+                name,
+                value,
+                type: 'Data',
+                timestamp: null,
+                submitter: contextValue.loc!.owner,
+                status: 'DRAFT'
+            }
+            dispatch({ type: 'ADD_ITEM', locItem: locItem })
+        }, [ contextValue.loc ]
+    )
+
+    const publishMetadataFunction = useCallback((item: LocItem) => {
+            const signAndSubmit: SignAndSubmit = (setResult, setError) => {
+                const callback: SignAndSendCallback = (signedTransaction) => {
+                    setResult(signedTransaction)
+                };
+                return addMetadata({
+                    locId: contextValue.locId,
+                    api: api!,
+                    signerId: item.submitter,
+                    item,
+                    callback,
+                    errorCallback: setError
+                })
+            };
+            return signAndSubmit;
+        }, [ api, contextValue.locId ]
+    )
+
+    const removeMetadataFunction = useCallback((locItem: LocItem) => {
+            dispatch({ type: 'DELETE_ITEM', locItem })
+        }, []
```

Comment:
Add `dispatch` to your dependency array.

Reply:
Done.